app/views/tasks/_form.html.haml

```diff
@@ -1,16 +1,23 @@
 =form_for @task do |f|
   -if @task.errors.any?
     #error_explanation
-      %h2="#{pluralize(@task.errors.count, "error")} prohibited this task from being saved:"
+      %h2= "#{pluralize(@task.errors.count, "error")} prohibited this task from being saved:"
       %ul
-        -@task.errors.full_messages.each do |msg|
+        - @task.errors.full_messages.each do |msg|
           %li= msg
 
   .field
-    =f.label :name
-    =f.text_field :name
+    = f.label :name
+    = f.text_field :name
+    = f.hidden_field :user_id, value: current_user.id
   .field
-    =f.label :completed
-    =f.check_box :completed
+    = f.fields_for @list do |ff|
+      %div
+        = ff.label :list
+        = ff.text_field :name
+  - if @task.persisted?
+    .field
+      = f.label :completed
+      = f.check_box :completed
```

Comment:
only display the `completed` checkbox within the form if the record has been saved into the db